src/Admin/OptionsMenu.php

```diff
@@ -440,27 +430,16 @@ private function list_template_conditional_options( $options, $parent = null ) {
 					continue;
 				}
 
+				$element_name = sprintf( '%s[supported_templates][%s]', AMP_Options_Manager::OPTION_NAME, rawurlencode( $id ) );
+				$supported    = ! empty( $supported_templates[ $id ] )
 				?>
 				<li>
-					<?php if ( empty( $option['immutable'] ) ) : ?>
-						<input
-							type="checkbox"
-							id="<?php echo esc_attr( $element_id ); ?>"
-							name="<?php echo esc_attr( $element_name ); ?>"
-							value="<?php echo esc_attr( $id ); ?>"
-							<?php checked( ! empty( $option['user_supported'] ) ); ?>
-						>
-					<?php else : // Persist user selection even when checkbox disabled, when selection forced by theme/filter. ?>
-						<input
-							type="checkbox"
-							id="<?php echo esc_attr( $element_id ); ?>"
-							<?php checked( ! empty( $option['supported'] ) ); ?>
-							<?php disabled( true ); ?>
-						>
-						<?php if ( ! empty( $option['user_supported'] ) ) : ?>
-							<input type="hidden" name="<?php echo esc_attr( $element_name ); ?>" value="<?php echo esc_attr( $id ); ?>">
-						<?php endif; ?>
-					<?php endif; ?>
+					<input
+						type="checkbox"
+						id="<?php echo esc_attr( $element_id ); ?>"
+						<?php checked( $supported ); ?>
+					>
+					<input type="hidden" name="<?php echo esc_attr( $element_name ); ?>" value="<?php echo esc_attr( wp_json_encode( ! empty( $supported_templates[ $id ] ) ) ); ?>">
```

Comment:
Why handle the checkbox value through a hidden input?

Reply:
It's because a checkbox that is not checked will not send any post data. So I modified the JS to toggle the hidden input with a boolean value. This is intended to be temporary since @johnwatkins0 will make it fully Reactive in #5015 .

Reply:
In other words, if no checkboxes are checked, then there will be no input vars to indicate the user is intending to make a change for these. This will not be an issue once moving to the REST API. I suppose my change here is overkill, however.

Reply:
Reverted/revised in 3a326c3b191d304022dda30d669c107f3da79fcc.